### DISCUSSION TITLE
Example of using --pre-run or CLI for setting parameters for path/query variables?

### DISCUSSION THREAD

**Discussion by dicolanl** [schemathesis/schemathesis#1772] [General]
Does anyone have an example of how to setup schemathesis cli to use some static variables in path or query parameters?

I create a hooks.py file:
```python
import schemathesis

@schemathesis.hooks.register
def before_call(context, case):
    if "/assets/{assetId}" in case.path:
        case.path_parameters["assetId"] = "myAsset"
```
then used --pre-run but it didnt like that.

```
An exception happened during the hook loading:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/schemathesis/cli/__init__.py", line 1022, in load_hook
    __import__(module_name)
ModuleNotFoundError: No module named 'hooks.py'; 'hooks' is not a package
```

Not quite sure how to do this.

**Comment by Stranger6667** [comment]
Sorry for the delay!

Schemathesis 3.24 has this feature built-in, without the need for hooks. Here are the [docs](https://schemathesis.readthedocs.io/en/stable/cli.html#overriding-test-data), let me know if you need any guidance on that.

On the hooks side. They should be loaded via the `SCHEMATHESIS_HOOKS` env var, the [docs](https://schemathesis.readthedocs.io/en/stable/cli.html#extending-cli) have been updated some time ago:

```
$ PYTHONPATH=$(pwd)
$ SCHEMATHESIS_HOOKS=test.setup
$ st run https://example.com/api/swagger.json
```

This will work if in the current directory has the `test` directory with `setup.py` in it. Let me know if that helps
